sdk/metric/internal/exemplar/rand.go

```diff
@@ -123,7 +123,9 @@ func (r *randRes) Offer(ctx context.Context, t time.Time, n Value, a []attribute
 	} else {
 		if r.count == r.next {
 			// Overwrite a random existing measurement with the one offered.
+			rngMu.Lock()
 			idx := int(rng.Int63n(int64(cap(r.store))))
+			rngMu.Unlock()
```

Comment:
All other access to this occurs with the lock held in the `random` function.

Reply:
If anyone could provide some guidance on where an appropriate test should be added I can give it a try. Unfortunately the bug only seems to trigger under a fair amount of load.

Reply:
Added a test which failed prior to the fix when run with `-race`.